File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
input = sys.stdin.readline

n = int(input())
m = int(input())

def dfs(x):
    visited[x] = True
    global cnt
    cnt += 1
 
    for i in graph[x]:
        if not visited[i]:
            dfs(i)

graph = [[] for _ in range(n+1)]
for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

visited = [False] * (n+1)
cnt = 0
dfs(1)
print(cnt-1)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Main {
    static boolean[] visited;
    static int[][] graph;
    static int N, M;
    static int cnt;

    public static void dfs(int x){
        visited[x] = true;
        cnt++;
        for(int i=0; i<=N; i++){
            if(graph[x][i] == 1 && !visited[i]){
                dfs(i);
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        M = Integer.parseInt(br.readLine());
        StringTokenizer st;
        graph = new int[N+1][N+1];
        for(int i=0; i<M; i++){
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            graph[a][b] = graph[b][a] = 1;
        }
        visited = new boolean[N+1];
        cnt = 0;
        dfs(1);
        System.out.println(cnt-1);
    }
}
